File: 111.minimum_depth_of_binary_tree/minDepth.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {
    if(root === null) return 0;
    var queue = [];
    var count = 0;
    queue.push(root);
    while(queue.length !== 0){
        var size = queue.length;//number of nodes in one level
        while(size-- > 0){
            var node = queue.shift(); 
            if (node.left === null && node.right === null){
                return count+1;
            }
            if(node.left !== null){
                queue.push(node.left);
            }
            if(node.right !== null){
                queue.push(node.right);
            }
        }
        count++;
    }
    return count;
};

//node.left !== null && node.right !== null

File: 111.minimum_depth_of_binary_tree/minDepth.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def minDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if root == None:
            return 0
        queue = []
        count = 0
        queue.append(root)
        while len(queue) != 0:
            #nodes in each level
            size = len(queue)
            while(size):
                node = queue.pop(0)
                if (node.left == None and node.right == None):
                    return count+1
                if (node.left != None):
                    queue.append(node.left)
                if (node.right != None):
                    queue.append(node.right)
                size -= 1
            count += 1

        return count